### PR TITLE
os configure: Locate the boot partition w/o using the device-type.json's partition field

### PR DESCRIPTION
**Pull request by thgreasi**
On top of https://github.com/balena-io/balena-cli/pull/2899, so let's get that in first for better debugging in case of an issue.

Update balena-device-init from 8.0.0 to 8.1.0

Depends-on: https://github.com/balena-io-modules/balena-device-init/pull/47
Change-type: patch

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -110,6 +110,24 @@ export async function getManifest(
 	const init = await import('balena-device-init');
 	const sdk = getBalenaSdk();
 	const manifest = await init.getImageManifest(image);
+	if (manifest != null) {
+		const config = manifest.configuration?.config;
+		if (config?.partition != null) {
+			const { getBootPartition } = await import('balena-config-json');
+			// Overwrite the deprecated  & no longer updated partition number on the device-type.json
```
Comment (kb2ma) [inline src/utils/helpers.ts:117]: It seems odd to sometimes update config.partition vs. config.partition.primary with the same information. Is config.partition.primary going away?

Comment (thgreasi) [reply src/utils/helpers.ts]: @kb2ma afaict both `partition: number` & `partition.primary: number` can be found in `/device-types/v1`. I have no clue whether anyone of these is going away. My intention is to check which of the two forms is used in the current DT and update it.
I updated the comment to make that a bit clearer.

**Review by kb2ma — COMMENTED** [review]
Nice tests! Not approving only due to package version on balena-device-init.

File: tests/commands/os/configure.spec.ts [hunk 2]
```diff
@@ -30,27 +32,175 @@ import { BalenaAPIMock } from '../../nock/balena-api-mock';
...
+			api.expectGetDeviceTypes();
+			// It should not reach to /config or /device-types/v1 but instead find
+			// everything required from the device-type.json in the image.
+			// api.expectGetConfigDeviceTypes();
+			api.expectDownloadConfig();
+
+			const command: string[] = [
+				`os configure ${tmpMatchingDtJsonPartitionPath}`,
+				'--device-type jetson-nano',
+				'--version 6.0.13',
```
Comment (kb2ma) [inline tests/commands/os/configure.spec.ts:109]: You can remove this line now that `getOsVersionFromImage()` reads from the image. It also would provide another implicit test.

File: tests/commands/os/configure.spec.ts [hunk 3]
```diff
@@ -30,27 +32,175 @@ import { BalenaAPIMock } from '../../nock/balena-api-mock';
...
+			api.expectGetDeviceTypes();
+			// It should not reach to /config or /device-types/v1 but instead find
+			// everything required from the device-type.json in the image.
+			// api.expectGetConfigDeviceTypes();
+			api.expectDownloadConfig();
+
+			const command: string[] = [
+				`os configure ${tmpNonMatchingDtJsonPartitionPath}`,
+				'--device-type jetson-nano',
+				'--version 6.1.25',
```
Comment (kb2ma) [inline tests/commands/os/configure.spec.ts:157]: You can remove this line now like for the preceding test.

**Comment by kb2ma** [comment]
### Testing
Testing jetson-xavier device type on staging.
#### v6.1.25
* OK, Download
* OK, device-type.json boot partition 43
* OK, Configure, including WiFi

#### v6.1.26
* OK, can't download, as expected
* OK, device-type.json boot partition 37
* OK, Configure, including WiFi

#### v3.1.3+rev1
* OK, Download
* OK, device-type.json boot partition 37
* OK, Configure with CLI PR and configure with CLI v20.0.0 -- both with WiFi

#### RPi 4, netio-logger-pi4 fleet on balena-cloud
* OK, Download
* OK, device-type.json boot partition 1
* OK, Configure with CLI PR

**Comment by kb2ma** [comment]
Retested configure of v6.1.26 with commit dbe177e, OK.

**Pull request merged**